### ISSUE TITLE
I2S_recorder IDF v4.4.2 sample rate off by 2x (IDFGH-8427)

### ISSUE THREAD

**Issue by davidallenmann** [espressif/esp-idf#9893]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v4.4.2

### Operating System used.

macOS

### How did you build your project?

Command line with CMake

### Development Kit.

ESP32-S3-DEVKIT-C1

### Power Supply used.

USB

### What is the expected behavior?

example should sample 15s audio and record to microSD

### What is the actual behavior?

Sample records PDM mono for 30 s but saves it as a 15 s file. So, when played back sound is pitch shifted 2x.
Changing the init_microphone function to multiply sample rate by 2 when setting up I2S clocks fixes issue.

```
void init_microphone(void)
{
    // Set the I2S configuration as PDM and 16bits per sample
    i2s_config_t i2s_config = {
        .mode = I2S_MODE_MASTER | I2S_MODE_RX | I2S_MODE_PDM,
        .sample_rate = CONFIG_EXAMPLE_SAMPLE_RATE * 2,
        .bits_per_sample = I2S_BITS_PER_SAMPLE_16BIT,
        .channel_format = I2S_CHANNEL_FMT_ONLY_LEFT,
        .communication_format = I2S_COMM_FORMAT_STAND_I2S,
        .intr_alloc_flags = ESP_INTR_FLAG_LEVEL2,
        .dma_buf_count = 8,
        .dma_buf_len = 200,
        .use_apll = 0,
    };

    // Set the pinout configuration (set using menuconfig)
    i2s_pin_config_t pin_config = {
        .mck_io_num = I2S_PIN_NO_CHANGE,
        .bck_io_num = I2S_PIN_NO_CHANGE,
        .ws_io_num = CONFIG_EXAMPLE_I2S_CLK_GPIO,
        .data_out_num = I2S_PIN_NO_CHANGE,
        .data_in_num = CONFIG_EXAMPLE_I2S_DATA_GPIO,
    };

    // Call driver installation function before any I2S R/W operation.
    ESP_ERROR_CHECK( i2s_driver_install(CONFIG_EXAMPLE_I2S_CH, &i2s_config, 0, NULL) );
    ESP_ERROR_CHECK( i2s_set_pin(CONFIG_EXAMPLE_I2S_CH, &pin_config) );
    ESP_ERROR_CHECK( i2s_set_clk(CONFIG_EXAMPLE_I2S_CH, CONFIG_EXAMPLE_SAMPLE_RATE * 2, I2S_BITS_PER_SAMPLE_16BIT, I2S_CHANNEL_MONO) );
}

```

### Steps to reproduce.

1. Open examples/peripherals/i2s/i2s_audio_recorder_sdcard
2. Build example and make a sample recording with PDM microphone (I used Vesper VM3000)
3. Note that a 15s recording actually takes 30s because the I2s clock rate is half of what it should be

**Comment by davidallenmann** [comment]
I believe this issue was supposed to have been addressed by #8660

**Comment by davidallenmann** [comment]
Thanks for the quick reply. I misunderstood that the release/ branch would have changes before the stable branch. Will check release/v4.4. Will reopen if have issues with that branch.
